client/my-sites/comments/comment-navigation/index.jsx

```diff
@@ -338,9 +350,15 @@ const mapStateToProps = ( state, { commentsPage, siteId } ) => {
 		}
 	} );
 
+	const dataRequests = get( state, 'dataRequests' );
+	const hasPendingBulkAction = filter( dataRequests, filterBulkActions ).reduce(
+		hasPendingActions,
+		false
+	);
```

Comment:
what's the purpose of the filter? could we not achieve the same result by doing this all in one `some()` and one iteration?

```js
some( dataRequests, ( { status }, type ) => ( type === COMMENTS… ) && status === 'pending' )
```

Reply:
actually I don't think this will work - did you test it? I'm pretty sure that the key in `state.dataRequests` is going to be a full serialization of the request, not just the action type

Reply:
We're just looking for related requests that we're interested in eg comment status change/delete. I guess I forgot about `some`, but at least when I tested it worked.

Reply:
Both approaches work.
I've tested both the `filter`+`reduce` and the `some` like this:
```es6
some(
	dataRequests,
	( { status }, action ) =>
		( includes( action, COMMENTS_CHANGE_STATUS ) || includes( action, COMMENTS_DELETE ) ) &&
		'pending' === status
)
```
They return the very same results, and in both cases they are correct.